actors/builtin/market/market_state.go

```diff
@@ -148,10 +148,32 @@ func (st *State) mutateDealProposals(rt Runtime, f func(*DealArray)) {
 	st.Proposals = rcid
 }
 
+func (st *State) mutateDealStates(rt Runtime, f func(*DealMetaArray)) {
+	states, err := AsDealStateArray(adt.AsStore(rt), st.States)
+	if err != nil {
+		rt.Abortf(exitcode.ErrIllegalState, "failed to load deal states array: %s", err)
+	}
```

Comment:
Would prefer not to add more uses of `rt` in state: return an error instead

Reply:
This is done.